### ISSUE TITLE
Depthpro has no Depth, especially with Depthpro L.

### ISSUE THREAD

**Issue by KolaKater** [nagadomi/nunif#503]
DepthPro doesn't seem to function reliably with iw3: depth details are often missing or incomplete, and occasionally no depth is generated at all.

This is the original image
![Image](https://github.com/user-attachments/assets/4f7a6285-7b04-4564-ba31-a8d203f9ab62)

it doesn't matter which setting for Depthpro Large I use, the result is the same like this:
![Image](https://github.com/user-attachments/assets/9dd189eb-c4a9-42fc-afbf-05b9c6d972bc)

**Comment by nagadomi** [comment]
It is probably because the depth is clipped at around 40 meters.
https://github.com/nagadomi/nunif/blob/562a8c589db2eaf9657973bbd01d5656b524ef01/iw3/depth_pro_model.py#L49

Try changing max_dist to 2000. In the official code, it is set to 250.
I was planning to make this an option, but I removed support for DepthPro at one point, so it has been left as a constant value.

**Comment by KolaKater** [comment]
Thank you Nagadomi. Based on the test of multiple pictures, 150 is a relatively good value at present. I will test the objects close to the lens after get off work.

**Comment by KolaKater** [comment]
after testing different images, it seems these values are good:
 def _forward(model, x, input_shape, min_dist=**_0.0_**, max_dist=**_150.0*_*): 
But there's still no depth for objects close to the lens.
for exmple: 
![Image](https://github.com/user-attachments/assets/c52ae774-a6c7-4a1d-87d9-f2077afb8d1d)

**Comment by nagadomi** [comment]
A thickness difference of about 15cm in a person is only about 1/100 of 150m in distance space, so the difference may be hard to see.
However, in disparity space, nearby differences appear larger.
Try foreground-scale 3.

Also, sometimes the difference can be seen in SBS mode.
This is because the display brightness has only 256 levels, but the calculations use float, so there can be details that are not visible directly on the display.

**Comment by KolaKater** [comment]
It seems here DepthPro is not for objects close to the lens. Only the background is allowed to have depth :-D
I am trying with different values, however the woman in depthmap and in SBS still looks like a piece of paper, even when I try foreground-scale 3 and max_dist=0.1

![Image](https://github.com/user-attachments/assets/7c2f593b-14b2-4436-a837-6203f9bfc919)
![Image](https://github.com/user-attachments/assets/22df20bf-5aa6-4921-b28d-33b7ef62d18e)

**Comment by nagadomi** [comment]
Then it might be an issue coming from the dataset preprocessing or the loss function used for metric depth training. I am not very familiar with metric depth training, but I can imagine that how the model handles sky distance can make a big difference.

It can also produce strange results when the input looks like a synthetic image. I once made a test video in Blender using IBL (skybox texture), but the depth estimation did not work well.

**Comment by KolaKater** [comment]
From my tests, one thing I like about DepthPro is that it detects object borders very well. I also tested DepthPro on Hugging Face, but the results weren’t any better. I’ve also seen tests from other users — they have the same issue: the depth maps generated by DepthPro show very little or almost no depth for foreground objects that are close to the camera. So, it’s very possible that DepthPro is not a very flexible or reliable depth map model

**Comment by KolaKater** [comment]
Just i saw and tested. [Here the Depthpro ](https://huggingface.co/spaces/hysts/DepthPro-transformers)has no problem:

![Image](https://github.com/user-attachments/assets/3af6183b-57a2-4026-919f-2676f48c5709)

**Comment by nagadomi** [comment]
Since both source codes are public, we should be able to find the differences.

https://huggingface.co/spaces/hysts/DepthPro-transformers/tree/main
https://github.com/huggingface/transformers/tree/main/src/transformers/models/depth_pro

**Comment by KolaKater** [comment]
It looks like depth_pro_model.py isn’t the only file linked to DepthPro on iw3 — other files may be involved too

**Comment by nagadomi** [comment]
Would setting min_dist=0.0001 and max_dist=10000 not work?
Another difference is that it visualizes 1 / metric_depth.

I think this could be solved quickly with a few tests, but I will not be able to try it until next month.

**Comment by KolaKater** [comment]
Thank you Nagadomi ❤️.
it's not really working with min_dist=0.0001 and max_dist=10000, but better for some other images than before.
From tests, the DepthPro Model L is currently the best-performing model, especially when it comes to rendering hair and sharp vertical lines or corners. If it works well on iw3, that would be great. It’s a bit slow, of course, but still produces very nice results for images

![Image](https://github.com/user-attachments/assets/dd455adc-d5a4-4a59-8c02-573a3df7d0e5)

**Comment by KolaKater** [comment]
I also tried this, but it is not working:

def _forward(model, x, input_shape, min_dist=0.0001, max_dist=150.0):
    if x.dtype != torch.float16:
        x = x.half()
    H, W = input_shape[2:]
    canonical_inverse_depth, fov_deg = model(x)
    canonical_inverse_depth = canonical_inverse_depth.to(torch.float32)

    if True:
        # distance
        f_px = 0.5 * W / torch.tan(0.5 * torch.deg2rad(fov_deg.to(torch.float)))
        inverse_depth = canonical_inverse_depth * (W / f_px)
        depth = 1.0 / torch.clamp(inverse_depth, min=1.0 / max_dist, max=1.0 / min_dist)
    else:
        # inverse distance
        f_px = 0.5 * W / torch.tan(0.5 * torch.deg2rad(fov_deg.to(torch.float)))
        inverse_depth = canonical_inverse_depth * (W / f_px)
        depth = torch.clamp(inverse_depth, min=1.0 / max_dist, max=1.0 / min_dist)

    if H < x.shape[2] or W < x.shape[3]:
        depth = F.interpolate(depth, size=(H, W), mode="bilinear", align_corners=False)

    # enhnace the front depth
    try:
        import cv2
        depth_np = depth.squeeze().cpu().numpy()
        depth_np = cv2.bilateralFilter(depth_np.astype(np.float32), d=9, sigmaColor=75, sigmaSpace=75)
        depth = torch.from_numpy(depth_np).unsqueeze(0).to(depth.device)
    except Exception:
        pass

    return depth

**Comment by nagadomi** [comment]
Fixed in https://github.com/nagadomi/nunif/commit/0e37df284906c7e609753a9e23a591fe08699e09 (both dev and iw3_inpaint branch)

<img width="1320" height="496" alt="Image" src="https://github.com/user-attachments/assets/2e01bbcd-2dd5-40b0-812f-43aa1c3bbe1c" />

Applied the same change to the VDA metric models.
No changes to DepthAnything1,2 or ZoeDepth metric models.
